### DISCUSSION TITLE
Solution for querying data using graphql

### DISCUSSION THREAD

**Discussion by 0xmonkeyking** [graphql/graphql-js#4149] [Q&A] [answered]
I need to fetch data for file directory with this format.
```
type File {
  id: ID!
  name: String!
  size: Int!
}

type Directory {
  id: ID!
  name: String!
  files: [File]
  directories: [Directory]
}

type Query {
  directory(id: ID!): Directory
}
```

How can I query data from using graphql client? Thanks in advance!

**Comment by JoviDeCroock** [comment]
There'a a lot of facets to your question, without the neccesary information it will be hard to get to the correct answer.

- Is this schema running on a server? If so you can use `fetch`/`graphiql`/any GraphQL client
- Is this a schema that you want to query? You can use the `execute` parameter this package exports

An example query for this schema would be

```graphql
query getDirectory($dir: ID!) {
  directory(id: $id) { id name }
}
```

In a fetch call this would look like

```js
fetch('your-url', {
  method: 'POST',
  body: JSON.stringify({ query, variables: { dir: 'x' } })
})
```

you can read more [here](https://graphql.org/graphql-js/graphql-clients/)

**Comment by 0xtomoon** [comment, marked as answer]
Unfortunately graphql client doesn't support recursive query for now.
Instead you can query the file directories in raw string and parse them in response.
